File: atcoder/abc066B_ss.ml
let even s =
    let n = String.length s in
    if n mod 2 != 0 then false
    else
        let first  = String.sub s 0 (n / 2) in
        let second = String.sub s (n / 2) (n / 2) in
        if first = second then true
        else false;;

let rec solution s =
    let n = String.length s in
    if even s then n
    else solution @@ String.sub s 0 (n - 1);;

Printf.printf "%d\n" @@
Scanf.scanf "%s\n" @@ fun s ->
    solution @@ String.sub s 0 @@ String.length s - 1;;

File: atcoder/abc066B_ss.nim
# vicfred
# https://atcoder.jp/contests/abc066/tasks/abc066_b
# string manipulation
import strutils

proc even(str: string): bool =
  if str.len mod 2 != 0:
    return false
  let size = str.len
  let first = str[0..<size div 2]
  let second = str[size div 2..<size]
  if first == second:
    return true
  return false

let s = stdin.readLine.strip

for i in 1..<s.len:
  if even(s[0..<s.len-i]):
    echo s.len-i
    quit()

